app/controllers/risks_controller.rb

```diff
@@ -68,14 +67,7 @@ def redirect_after_update
   end
 
   def form_params
-    params[step]
-  end
-
-  def form
-    @_form ||= form_for_risk_section(step).new(risk)
-  end
-
-  def form_for_risk_section(section)
-    "Forms::Risk::#{section.to_s.camelcase}".constantize
+    return unless params[step]
+    params[step].permit!.to_h
```

Comment:
📓 The `permit!.to_h` was introduced here only as a temporary measure to avoid a _**deprecation warning**_ related with the `ActionController::Parameters` and the use of `.with_indifferent_access` on it.

Ideally we should only passed the `permitted` params, but given this PR is introducing a dynamic way of populating the form, supporting the `permitted` params would imply having to add some more methods to the form to returned the `permitted` fields.

That might come in an upcoming PR. 

Also, this is not a major issue at all, as the form by a `reduce` strategy only updates the fields described in the _section schema_.